packages/dids/src/dht.ts

```diff
@@ -0,0 +1,312 @@
+        ttl  : TTL,
+        data : `id=${sId},t=${service.type},uri=${service.serviceEndpoint}`
+      };
+
+      packet.answers.push(serviceRecord);
+      svcIds.push(recordIdentifier);
+    });
+
+    // add root record for vms and svcs
+    if (vmIds.length) {
+      rootRecord.push(`vm=${vmIds.join(',')}`);
+    }
+    if (svcIds.length) {
+      rootRecord.push(`svc=${svcIds.join(',')}`);
+    }
+
+    // add verification relationships
+    if (document.authentication) {
+      const authIds: string[] = document.authentication
+        .map(id => DidDht.identifierFragment(id))
+        .filter(id => keyLookup.has(id))
+        .map(id => keyLookup.get(id) as string);
+      if (authIds.length) {
+        rootRecord.push(`auth=${authIds.join(',')}`);
+      }
+    }
+    if (document.assertionMethod) {
+      const authIds: string[] = document.assertionMethod
+        .map(id => DidDht.identifierFragment(id))
+        .filter(id => keyLookup.has(id))
+        .map(id => keyLookup.get(id) as string);
+      if (authIds.length) {
+        rootRecord.push(`asm=${authIds.join(',')}`);
+      }
+    }
+    if (document.keyAgreement) {
+      const authIds: string[] = document.keyAgreement
+        .map(id => DidDht.identifierFragment(id))
+        .filter(id => keyLookup.has(id))
+        .map(id => keyLookup.get(id) as string);
+      if (authIds.length) {
+        rootRecord.push(`agm=${authIds.join(',')}`);
+      }
+    }
+    if (document.capabilityInvocation) {
+      const authIds: string[] = document.capabilityInvocation
+        .map(id => DidDht.identifierFragment(id))
+        .filter(id => keyLookup.has(id))
+        .map(id => keyLookup.get(id) as string);
+      if (authIds.length) {
+        rootRecord.push(`inv=${authIds.join(',')}`);
+      }
+    }
+    if (document.capabilityDelegation) {
+      const authIds: string[] = document.capabilityDelegation
+        .map(id => DidDht.identifierFragment(id))
+        .filter(id => keyLookup.has(id))
+        .map(id => keyLookup.get(id) as string);
+      if (authIds.length) {
+        rootRecord.push(`del=${authIds.join(',')}`);
+      }
+    }
+
+    // Add root record
+    packet.answers.push({
+      type : 'TXT',
+      name : '_did',
+      ttl  : TTL,
+      data : rootRecord.join(';')
+    });
+
+    return packet as Packet;
+  }
+
+  /**
+     * Extracts the fragment from a DID
+     * @param identifier The DID to extract the fragment from
+     */
+  private static identifierFragment(identifier: string): string {
+    return identifier.includes('#') ? identifier.substring(identifier.indexOf('#') + 1) : identifier;
+  }
+
+  /**
+     * Retrieves a DID Document from the DHT
+     * @param did The DID of the document to retrieve
+     */
+  public static async getDidDocument(did: string): Promise<DidDocument> {
+    const didFragment = did.replace('did:dht:', '');
+    const publicKeyBytes = new Uint8Array(z32.decode(didFragment));
+    for (const relay of PKARR_RELAYS) {
```

Comment:
@decentralgabe should the pkarr relay used to retrieve the DID doc be a configurable parameter?  Fine with addressing this in a future PR.

Reply:
same comment as above

Reply:
updated to support a configurable relay with a default set by the lib